### ISSUE TITLE
Copy to a mounted cifs share fails because of attempt to change file mode

### ISSUE THREAD

**Issue by StefanHirche** [gradle/gradle#22587]
### Expected Behavior

The files should be copied to the destination folder without trying to change the file mode afterwards

### Current Behavior

After copying a file, gradle tries to change the file mode. This fails, as it is not allowed.

### Context

I'd like to copy files to a cifs share. This is not possible. I even used fileMode=null and dirMode=null, still gradle tries do change the fileMode to 0664. Also changing the fileMode to 0777 does not change the behaviour. But changing the file mode is not allowed at all.

### Steps to Reproduce 

Basically mount a windows share with cifs on a linux machine and then try to copy or sync files with a gradle task to this mount.
I have to mention, the zip task does work, so I can create a zip file on the share, as this does not try to change the file mode afterwards.

### Your Environment

Build scan URL:

**Comment by StefanHirche** [comment]
OK, there seems to be a workaround. If you add 'noperm' as a mount option, the chmod is allowed but ignored.

Still it would be great to have an option to not change the file mode at all.

**Comment by jbartok** [comment]
Sorry that you're having trouble with Gradle!

Your issue lacks information about how to reproduce the problem you're having. A reproducer project can really help us track down and fix your problem quicker.  We may also be able to suggest workarounds or ways to avoid the problem if we can reproduce it.

You can use the following as a base for your reproducer: https://github.com/gradle/gradle-issue-reproducer

This issue will be closed after 7 days, unless you can provide more information.
